### PR TITLE
Add ozans playlist concept exercise

### PR DESCRIPTION
**Pull request by kristinaborn**
@SleeplessByte I'd love your thoughts on what I've got so far. It was surprisingly difficult to come up with a story for this one.

Hopefully I followed all the new guidelines correctly 🤞

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Sweet. I've invited @junedev to review as well as she has a great mental map of the concept-tree as we want it. 

As for the failing CI check, here is how you can resolve it (there are more efficient ways to do this, but I like to take the following steps so I know _exactly_ what's going on):

First bring your `main` branch up-to-date:

1. `git fetch upstream`
2. `git checkout main`
3. `git pull`
4. `git checkout add-ozans-playlist-concept-exercise`

Then sync it:

5. `git rebase main`
6. `npx babel-node scripts/sync`
7. `git add exercises/concepts/sets`
8. `git commit -m "Sync all files"`
9. `git push --force`

I will attempt to review it before Friday, but chat me up on Slack if you want faster feedback.

**Comment by SleeplessByte** [comment]
(That triggered because the `scripts` `sync` _contains_ the command to run the `sync` action. Obviously if it all turns green now, you don't need to do anything in that regard).

**Review by junedev — COMMENTED** [review]
I did not do a line-by-line review yet but from what I could glance this looks like great work. :slightly_smiling_face: 

I added some general comments that might be helpful.

File: concepts/sets/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "blurb": "TODO: add blurb for sets concept",
```
[the remark this answers is not in the document]
Comment (kristinaborn) [reply concepts/sets/.meta/config.json]: Thanks for the hint! Added the blurb.

File: concepts/sets/about.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# About
+
```
[the remark this answers is not in the document]
Comment (kristinaborn) [reply concepts/sets/about.md:2]: Makes sense! Moved the overall MDN link to the links file and removed the set constructor link.

File: concepts/sets/.meta/config.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (kristinaborn) [reply concepts/sets/.meta/config.json:1]: Done :)

File: exercises/concept/ozans-playlist/.meta/design.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+The Concepts this exercise unlocks are:
+
+- `sets`
+
+## Prerequisites
+
+- `array-destructuring` because examples use array destructuring and the spread operator
+- `equality` because it is referenced in the introduction and `about.md`
+- `strings` because one task involves splitting a string
```
[the remark this answers is not in the document]
Comment (kristinaborn) [reply exercises/concept/ozans-playlist/.meta/design.md]: Updated prerequisites in both files.

**Comment by kristinaborn** [comment]
Marking this as ready to review, but further suggestions are certainly welcome!

File: exercises/concept/ozans-playlist/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+## 1. Remove duplicate tracks
+
+Implement the `removeDuplicates` function, which takes a playlist as a _parameter_ and _returns_ a new playlist where all the tracks are unique.
+
+```javascript
+const playlist = [
+  'Court and Spark - Joni Mitchell',
+  'Big Yellow Taxi - Joni Mitchell',
+  'Court and Spark - Joni Mitchell
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/.docs/instructions.md:17]:
```suggestion
  'Court and Spark - Joni Mitchell'
```

**Review by SleeplessByte — APPROVED** [review]

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  addTrack,
+  deleteTrack,
+  hasTrack,
+  listArtists,
+  removeDuplicates,
+} from './ozans-playlist';
+
+describe("Ozan's playlist", () => {
+  describe('removeDuplicates', () => {
+    xtest('works for an empty playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:11]:
```suggestion
    test('works for an empty playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [];
+
+      expect(removeDuplicates(playlist)).toEqual([]);
+    });
+
+    xtest('works for a non-empty playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:17]:
```suggestion
    test('works for a non-empty playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+      expect(removeDuplicates(playlist)).toEqual(expected);
+    });
+  });
+
+  describe('hasTrack', () => {
+    const TRACK_1 = 'Big Science - Laurie Anderson';
+    const TRACK_2 = 'Tightrope - Laurie Anderson';
+
+    xtest('returns true when the track is in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:31]:
```suggestion
    test('returns true when the track is in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [TRACK_1, TRACK_2];
+
+      expect(hasTrack(playlist, TRACK_1)).toBe(true);
+    });
+
+    xtest('returns false when the track is not in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:37]:
```suggestion
    test('returns false when the track is not in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+      expect(hasTrack(playlist, TRACK_1)).toBe(false);
+    });
+  });
+
+  describe('addTrack', () => {
+    const TRACK_1 = 'Jigsaw Feeling - Siouxsie and the Banshees';
+    const TRACK_2 = 'Feeling Good - Nina Simone';
+
+    xtest('adds a track that is not already in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:48]:
```suggestion
    test('adds a track that is not already in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [];
+      const expected = [TRACK_1];
+
+      expect(addTrack(playlist, TRACK_1)).toEqual(expected);
+    });
+
+    xtest('does not add a track that is already in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:55]:
```suggestion
    test('does not add a track that is already in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+      expect(addTrack(playlist, TRACK_1)).toEqual(expected);
+    });
+  });
+
+  describe('deleteTrack', () => {
+    const TRACK_1 = 'Ancestors - Tanya Tagaq';
+    const TRACK_2 = 'Take This Hammer - Lead Belly';
+
+    xtest('works if the track is present in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:67]:
```suggestion
    test('works if the track is present in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [TRACK_1, TRACK_2];
+      const expected = [TRACK_2];
+
+      expect(deleteTrack(playlist, TRACK_1)).toEqual(expected);
+    });
+
+    xtest('works if the track is not present in the playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:74]:
```suggestion
    test('works if the track is not present in the playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [TRACK_2];
+      const expected = [TRACK_2];
+
+      expect(deleteTrack(playlist, TRACK_1)).toEqual(expected);
+    });
+  });
+
+  describe('listArtists', () => {
+    xtest('works for an empty playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:83]:
```suggestion
    test('works for an empty playlist', () => {
```

File: exercises/concept/ozans-playlist/ozans-playlist.spec.js [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+      const playlist = [];
+
+      expect(listArtists(playlist)).toEqual([]);
+    });
+
+    xtest('works for a non-empty playlist', () => {
```
Comment (SleeplessByte) [inline exercises/concept/ozans-playlist/ozans-playlist.spec.js:89]:
```suggestion
    test('works for a non-empty playlist', () => {
```

**Comment by SleeplessByte** [comment]
Thank you @junedev for all the wise words, and thank you @kristinaborn for this exercise. Let's merge it now and if we need changes make it in future PRs.

**Comment by SleeplessByte** [comment]
@kristinaborn your exercise doesn't show up yet, likely because `sets` doesn't exist in `config.json` as `concept`. If you can do a second PR to fix that, that'd be great!

**Pull request merged**